plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/TestBloomFilter.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.parquet;
+
+import com.google.common.io.Resources;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.trino.filesystem.TrinoFileSystem;
+import io.trino.filesystem.TrinoFileSystemFactory;
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.hdfs.HdfsFileSystemFactory;
+import io.trino.parquet.BloomFilterStore;
+import io.trino.parquet.ParquetReaderOptions;
+import io.trino.parquet.reader.MetadataReader;
+import io.trino.plugin.hive.FileFormatDataSourceStats;
+import io.trino.plugin.hive.HiveConfig;
+import io.trino.plugin.hive.HiveStorageFormat;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.type.VarcharType;
+import org.apache.parquet.column.values.bloomfilter.BloomFilter;
+import org.apache.parquet.hadoop.metadata.ColumnChunkMetaData;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Optional;
+
+import static io.trino.parquet.predicate.TupleDomainParquetPredicate.getBloomFilterHash;
+import static io.trino.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.trino.plugin.hive.HiveTestUtils.getHiveSession;
+import static io.trino.spi.type.VarcharType.createVarcharType;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestBloomFilter
+{
+    public static final ConnectorSession SESSION = getHiveSession(new HiveConfig().setHiveStorageFormat(HiveStorageFormat.PARQUET));
+
+    @Test
+    public void testReadBloomFilterParquet()
+            throws Exception
+    {
+        String[] testNames = {"hello", "parquet", "bloom", "filter"};
+        VarcharType columnType = createVarcharType(255);
+
+        File parquetFile = new File(Resources.getResource("bloom-filter.parquet").toURI());
```

Comment:
Does the `bloom-filter.parquet` file need to be `1MB` ?

Reply:
yes, it is the default value.

Reply:
From the parquet-mr code:

  int maxBloomFilterSize = props.getMaxBloomFilterBytes();

    OptionalLong ndv = props.getBloomFilterNDV(path);
    OptionalDouble fpp = props.getBloomFilterFPP(path);
    // If user specify the column NDV, we construct Bloom filter from it.
    if (ndv.isPresent()) {
      int optimalNumOfBits = BlockSplitBloomFilter.optimalNumOfBits(ndv.getAsLong(), fpp.getAsDouble());
      this.bloomFilter = new BlockSplitBloomFilter(optimalNumOfBits / 8, maxBloomFilterSize);
    } else {
      this.bloomFilter = new BlockSplitBloomFilter(maxBloomFilterSize);
    }

However, in Trino, when implementing write path, we can definitely specify a different size.

Reply:
maxBloomFilterSize is 1mb